sample-apps/go-sample-app/collection/client.go

```diff
@@ -90,28 +97,25 @@ func StartClient(ctx context.Context) (func(context.Context) error, error) {
 			}
 		}()
 		// pushes any last exports to the receiver
-		err = ctrl.Stop(ctx)
+		mpErr := meterProvider.Shutdown(ctx)
+		if mpErr != nil {
+			err = mpErr
+		}
 		return
 	}, nil
 }
 
 // setupTraceProvider configures a trace exporter and an AWS X-Ray ID Generator.
-func setupTraceProvider(ctx context.Context) (*sdktrace.TracerProvider, error) {
+func setupTraceProvider(ctx context.Context, res *resource.Resource) (*sdktrace.TracerProvider, error) {
 	// INSECURE !! NOT TO BE USED FOR ANYTHING IN PRODUCTION
-	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure(),
-		otlptracegrpc.WithEndpoint(grpcEndpoint),
-		otlptracegrpc.WithDialOption(grpc.WithBlock()))
+	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure())
```

Comment:
Why are the endpoint and blocking connect options removed?

Reply:
With the blocking connect option, the sample app does not start when I run it against the new testcase in the test framework locally.  I assume because the sample app is blocking until the collector connects to it after the sample app calls, but runs endlessly if the collector starts before the sample app.  I think the default behavior of the dial returning immediately and starting the app is better suited for all situations.  With the endpoint option, if `WithEndpoint` is not specified then the default behavior is to set the endpoint to `localhost:4317` which pass the tests in the test framework.  Setting the endpoint to `0.0.0.0:4317` also make the sample app fail to export to the collector so I think `localhost:4317` works better for the situation, but please correct me if I'm wrong.